src/game.cpp

```diff
@@ -411,7 +411,7 @@ void game::init_ui()
         locW = sidebarWidth;
         stat2H = 2;
         stat2W = sidebarWidth;
-        messH = TERMY - (statH + locH + stat2H);
+        messH = TERRAIN_WINDOW_HEIGHT - (statH + locH + stat2H);
```

Comment:
TERMY and TERRAIN_WINDOW_HEIGHT use a different coordinate system. I changed my "map_fontheight" setting to 4 (yes, it's unreadable, but that's not the point) and kept the main "fontheight" at 16. The messages (and other text) is printed with height 16, but the terrain window uses the height 4. This results in TERRAIN_WINDOW_HEIGHT being 121 (the absolute maximum) and TERMY being 40 (directly taken from the options, I think).

As a result, the message window is *far* to high and the first message is printed *far* below the visible area of the screen (and as such the first few messages are not visible at all).

TERRAIN_WINDOW_* measures the number of rows/columns of the terrain window, one row/column of it does not necessarly equal one row/column of the main window/the message window.

I think you need to translate them into the main coordinate system with `from_map_font_dimension`, something like this does it for me:
```C++
{
    // Sorry for the meaningless variables name, I'm currently out of creativity.
    int dummy[2] = { 1, TERRAIN_WINDOW_HEIGHT };
    from_map_font_dimension( dummy[0], dummy[1] );
    // Replace TERRAIN_WINDOW_HEIGHT with dummy[1]:
    mouseview_h = dummy[1] - mouseview_y - 5;
}
```